web/pdf_page_view.js

```diff
@@ -807,6 +812,7 @@ class PDFPageView {
         }
         this.eventBus._off("textlayerrendered", this._onTextLayerRendered);
         this._onTextLayerRendered = null;
+        this._accessibilityManager.onTextLayerRendered();
```

Comment:
Most likely this line, or rather its placement, explains the problem mentioned in https://github.com/mozilla/pdf.js/pull/15237#issuecomment-1200655671.

By placing this line here you've limited this functionality to *only* documents with structureTree-information, which I'm guessing isn't actually intended?
If you actually want to call this method *unconditionally*, you'll need to pass in the `AccessibilityManager`-instance to the `TextLayerBuilder`-instance and invoke that method there instead (similar to how the `TextHighlighter`-instance is handled).

Basically, I'm guessing that you want to call it *after* the following line instead: https://github.com/mozilla/pdf.js/blob/803e7af5954b8d84f8dab2b632a3167c7fd14e46/web/text_layer_builder.js#L116

Reply:
> I took a look at this, but I wasn't able to see the annotations with a screen reader. Some questions:
> 
>     1. Perhaps I'm not understanding how to commit/save annotations to a document. I made a text annotation, dismissed the editor with control+enter, saved the PDF with control+s and then opened the saved PDF in the modified pdf.js. Is there anything else I need to do? I'm pretty sure I can see the annotation with OCR, but it's hard to be certain.
> 
>     2. Is there anything else I need to do when switching pdf.js branches apart from npm install and gulp server?
>

It's exactly what you've to do.
 
>     3. Are you able to see the annotations in the Firefox Accessibility Inspector?
> 
> 
> Also, how will this interact with tagged PDF? I don't fully understand how the text and structure layers fit together, but I know tagged PDF (structure) uses aria-owns as well.

I fixed the bug mentioned by Jonas and it should be fine now: I can see it correctly in the inspector.
Anyway, for now the content of the annotation is still not available: I just added an `aria-owns` between a span in the text layer and an annotation in the annotation layer. So it should be useful to see at least the annotations which don't have any content like a text field or a checkbox.
I've a wip to make the text content of a FreeText annotation in its HTML counterpart.


Ab

Reply:
> Anyway, for now the content of the annotation is still not available: I just added an aria-owns between a span in the text layer and an annotation in the annotation layer. So it should be useful to see at least the annotations which don't have any content like a text field or a checkbox.

I'm not really sure how to test this/provide feedback if text annotations aren't fully exposed yet, as I can't create other kinds of annotations.

Reply:
Once this PR is merged (it should be in 30 minutes):
https://github.com/mozilla/pdf.js/pull/15267

it should be a way easier to test this PR.

Reply:
@jcsteh, I merged the other PR and I rebased this one, so we should be good now.
Sorry for the inconvenience: I should have written the second patch in first.